diff.py

```diff
@@ -1880,21 +1950,136 @@ def progress(self, msg: str) -> None:
+            ("localhost", self.config.http_server_port), MyReqHandler
+        )
+        self.running = True
+        # fixme
+        print(f"Http server is running on port {self.config.http_server_port}")
+        print("Open with ?once (eg file:///.../client.html?once)")
+        if self.config.run_browser:
+            if self.config.run_browser_command:
+                os.system(self.config.run_browser_command.format(query="once"))
+            else:
+                print("--browse was passed but 'browser' isn't set in diff_settings.py")
+        self.http_server.serve_forever()
+
+    def run_async(self, watch_queue: "queue.Queue[Optional[float]]") -> None:
+        self.watch_queue = watch_queue
+        self.ready_queue = queue.Queue()
+
+        web_display = self
+
+        class MyReqHandler(http.server.BaseHTTPRequestHandler):
+            def do_GET(self):
+                if not web_display.running:
+                    return
+                self.send_response(http.HTTPStatus.OK)
+                self.send_header("Access-Control-Allow-Origin", "*")
+                self.end_headers()
+                web_display.ready_queue.get()
+                self.wfile.write(web_display.run_diff().encode("utf-8"))
+                self.wfile.flush()
+
+        self.http_server = http.server.HTTPServer(
+            ("localhost", self.config.http_server_port), MyReqHandler
+        )
+        self.running = True
+        threading.Thread(target=self.http_server.serve_forever).start()
+        # fixme sync better
+        print(f"Http server is running on port {self.config.http_server_port}")
+        if self.config.run_browser:
+            if self.config.run_browser_command:
+                os.system(self.config.run_browser_command.format(query=""))
+            else:
+                print("--browse was passed but 'browser' isn't set in diff_settings.py")
+        self.ready_queue.put(None)
+
+    def progress(self, msg: str) -> None:
+        # fixme
+        print(msg)
+
+    def update(self, mydump: str) -> None:
+        if mydump == self.mydump:
+            self.progress("Unchanged. ")
             return
-        self.less_proc.kill()
-        self.ready_queue.get()
+        self.mydump = mydump
+        self.ready_queue.put(None)
+
+    def update_error(self, error: str) -> None:
+        # fixme
+        print("Error:")
+        print(error)
 
     def terminate(self) -> None:
-        if not self.less_proc:
+        self.running = False
+        self.ready_queue.put(None)
+        self.http_server.shutdown()
+
+
+class ConsoleDisplay(Display):
+    def print_diff(self) -> None:
+        print(self.run_diff())
+
+    def run_sync(self) -> None:
+        self.print_diff()
+
+    def run_async(self, watch_queue: "queue.Queue[Optional[float]]") -> None:
+        self.print_diff()
+
+    def progress(self, msg: str) -> None:
+        print(msg)
+
+    def update(self, mydump: str) -> None:
```

Comment:
is this ever run? seems weird to have watch but no pager

Reply:
it is and it works, if you run --watch --no-pager you get the diff spit out in stdout. Not usable by any mean but if you want it it's there
the main code (that does the refreshing if --watch) is Display-subclass-independent, if that makes sense